test/harness/e2e/harness.go

```diff
@@ -935,3 +1296,42 @@ func ConditionExists(d *v1alpha1.Device, conditionType, conditionStatus, conditi
 	}
 	return false
 }
+
+func (h *Harness) WaitForFleetUpdateToFail(fleetName string) error {
+	logrus.Infof("Waiting for fleet update to fail for fleet %s", fleetName)
+	Eventually(func() (bool, error) {
+		rolloutStatus, err := h.GetRolloutStatus(fleetName)
+		if err != nil {
+			return false, err
+		}
+
+		if rolloutStatus.Type == v1alpha1.FleetRolloutInProgress &&
+			rolloutStatus.Status == v1alpha1.ConditionStatusFalse &&
+			rolloutStatus.Reason == v1alpha1.RolloutSuspendedReason {
+			logrus.Infof("Fleet update failed for fleet %s: status=%s, reason=%s",
+				fleetName, rolloutStatus.Status, rolloutStatus.Reason)
+			return true, nil
+		}
+
+		return false, nil
+	}, LONGTIMEOUT, POLLING).Should(BeTrue(),
+		fmt.Sprintf("Timed out waiting for fleet %s update to fail", fleetName))
+	return nil
+}
+
+func (h *Harness) WaitForDeviceUpdateToSucceed(deviceId string) (err error) {
```

Comment:
THis functionality is implemented in WaitForDeviceNewRenderedVersion() function.
Could you please use  it, or refactor its "WaitForDeviceContents" part with your function